sphinx-jsonschema/__init__.py

```diff
@@ -301,14 +301,17 @@ def from_url(self, url):
         data = response.content.decode()
         return data, url
 
-    def from_file(self, filename):
+    def _convert_filename(self, filename):
+        """
+        Join the filename with the docs' source if it is not absolute.
+        """
+        if os.path.isabs(filename):
+            return filename
+        # file relative to the path of the current rst file
         document_source = os.path.dirname(self.state.document.current_source)
-        if not os.path.isabs(filename):
-            # file relative to the path of the current rst file
-            source = os.path.join(document_source, filename)
-        else:
-            source = filename
+        return os.path.join(document_source, filename)
 
+    def from_file(self, source):
```

Comment:
In contrast to its previous functionality, `from_file` now requires its argument to be an absolute path.

Also renaming the `filename` argument to `source` introduces a breaking change that is not necessary. I suggest you keep the `filename` argument as is and call
```python
source = self._convert_filename(filename)
```
at the method's beginning.

Reply:
See 36efc5e16246a2a65a00230ec3cf5b4334be1a6a, 97c8d282b6252c27691702b8705f281d408acdc2